cmd/source_profile.go

```diff
@@ -0,0 +1,283 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+)
+
+type SourceProfileType int
+
+const (
+	SourceProfileTypeUnset = iota
+	SourceProfileTypeFile
+	SourceProfileTypeConnection
+	SourceProfileTypeConfig
+)
+
+type SourceProfileFile struct {
+	path   string
+	format string
+}
+
+func NewSourceProfileFile(params map[string]string) SourceProfileFile {
+	profile := SourceProfileFile{}
+	if !filePipedToStdin() {
+		profile.path = params["file"]
+	}
+	if format, ok := params["format"]; ok {
+		profile.format = format
+		// TODO: Add check that format takes values from ["dump", "csv", "avro", ... etc]
+	} else {
+		fmt.Printf("source-profile format defaulting to `dump`\n")
+		profile.format = "dump"
+	}
+	return profile
+}
+
+type SourceProfileConnectionType int
+
+const (
+	SourceProfileConnectionTypeUnset = iota
+	SourceProfileConnectionTypeMySQL
+	SourceProfileConnectionTypePostgreSQL
+	SourceProfileConnectionTypeDynamoDB
+)
+
+type SourceProfileConnectionMySQL struct {
+	host string // Same as MYSQLHOST environment variable
+	port string // Same as MYSQLPORT environment variable
+	user string // Same as MYSQLUSER environment variable
+	db   string // Same as MYSQLDATABASE environment variable
+	pwd  string // Same as MYSQLPWD environment variable
+}
+
+func NewSourceProfileConnectionMySQL(params map[string]string) SourceProfileConnectionMySQL {
+	// TODO: Move parsing of environment variables in this function.
+	mysql := SourceProfileConnectionMySQL{}
+	if host, ok := params["host"]; ok {
+		mysql.host = host
+	}
+	if port, ok := params["port"]; ok {
```

Comment:
should set the default for stuff like port: 3306. Which rarely change.
Another option is to even default host to localhost

Reply:
Done.